src/visitor.ts

```diff
@@ -147,6 +155,22 @@ abstract class Visitor {
         this.visit(node.body);
     }
 
+    protected visitForStatement(node: ForStatement) {
```

Comment:
I don't know if this should be the default behaviour. Wouldn't it be more consistent if you visit the `update` before visiting the `body` here?

Reply:
It doesn't really matter, but it simplifies the generator visitor logic, because I need to be able to pop the `update` without poping the `body`. Otherwise I would need to store the `body` in a local variable and pass it to the `createLoop` helper method.